src/js/other-websites/icon.js

```diff
@@ -0,0 +1,452 @@
+    invisibleFocusableButton.style.border = "0px none";
+    invisibleFocusableButton.style.clip = "rect(0px, 0px, 0px, 0px)";
+    invisibleFocusableButton.style.clipPath = "inset(50%)";
+    invisibleFocusableButton.style.height = "1px";
+    invisibleFocusableButton.style.margin = "0px -1px -1px 0px";
+    invisibleFocusableButton.style.overflow = "hidden";
+    invisibleFocusableButton.style.padding = "0px";
+    invisibleFocusableButton.style.position = "absolute";
+    invisibleFocusableButton.style.width = "1px";
+    invisibleFocusableButton.style.whiteSpace = "nowrap";
+    invisibleFocusableButton.addEventListener("focus", () => {
+      applyIconFocusStyle(input);
+    });
+    invisibleFocusableButton.addEventListener("blur", () => {
+      applyIconBlurStyle(input);
+    });
+    invisibleFocusableButton.addEventListener("click", (event) => {
+      event.preventDefault();
+      openMenu(input);
+    });
+    input.insertAdjacentElement("afterend", invisibleFocusableButton);
+  }
+
+  sendRelayEvent("In-page", "input-icon-injected", "input-icon");
+}
+
+/**
+ * @param {HTMLInputElement} element 
+ * @returns number
+ */
+function getPaddingRight(element) {
+  const elementStyles = getComputedStyle(element);
+  return Number.parseInt(elementStyles.paddingRight.replace("px", ""), 10);
+}
+
+/**
+ * @param {HTMLInputElement} element 
+ * @param {{ x: number }} coords 
+ * @returns boolean
+ */
+function intersectsRelayIcon(
+  element,
+  coords
+) {
+  return element.clientWidth - coords.x <= 25 + 2 * getPaddingRight(element);
+}
+
+/**
+ * @param {MouseEvent} event 
+ * @returns void
+ */
+function onEmailInputClick(event) {
+  const target = event.target;
+  if (!(target instanceof HTMLInputElement)) {
+    return;
+  }
+  if (!intersectsRelayIcon(target, { x: event.offsetX })) {
+    return;
+  }
+
+  // Do not focus the input field
+  event.preventDefault();
+  openMenu(target);
+}
+
+/**
+ * @param {HTMLInputElement} target 
+ * @returns void
+ */
+function openMenu(target) {
+  sendRelayEvent("In-page", "input-icon-clicked", "input-icon");
+  // TODO: Trap focus inside the underlay.
+  //       See e.g. https://css-tricks.com/a-css-approach-to-trap-focus-inside-of-an-element/
+  const existingUnderlay = document.getElementById("relay-popover-underlay");
+  const underlay = existingUnderlay ?? document.createElement("div");
+  underlay.id = "relay-popover-underlay";
+  underlay.style.display = "initial";
+  underlay.style.position = "fixed";
+  underlay.style.inset = "0";
+  underlay.style.zIndex = "99999999";
+  underlay.style.backgroundColor = "rgba(0, 0, 0, 0.1)";
+  const existingIframe = underlay.querySelector("iframe");
+  const iframe = existingIframe ?? document.createElement("iframe");
+  iframe.src = browser.runtime.getURL("inpage-panel.html");
+  iframe.style.position = "absolute";
+  iframe.style.width = `${RELAY_INPAGE_MENU_WIDTH}px`;
+  iframe.style.maxWidth = `95vw`;
```

Comment:
Per my comment at the top of this review, I expect how the iframe is being initialized is where we would set the tip/rounded corners. 

<img width="379" alt="image" src="https://user-images.githubusercontent.com/2692333/216411790-855f4951-d2a1-42e2-a15c-39224d2f1be8.png">

Reply:
Hmm crap, in my spike I added this icon inside the iframe. I'll have to do some rework to be able to add it here, since you can't add `::before` elements (or other children) to an iframe, so all the positioning will have to be redone to apply to a wrapper div. Hopefully I can find some time for that next week.

(Btw, this PR does already have rounded corners, and as far as I can see they're the same size -4px- as the current implementation?)

Reply:
OK, I've struggled with making this appear properly for a bit now (in particular, getting it to take up the right space and have a consistent size), and I can't seem to make it work. My proposal would be to go without the chevron for now - it's relatively clear that the icon is related to the icon, so I think not having it for now would be better than having the current buggy icon implementation. I'm going to ask QA to test this.